### PR TITLE
[dev] Update Filter drawer modal to handle back button

### PR DESCRIPTION
**Pull request by krishnakant-technossus**
## Filter Drawer 

- Added `onRequestClose` prop to handle the back button and close the modal.

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -500,6 +500,13 @@ export const App: React.FC<IProps> = (): React.ReactElement => {
                 </Button>
 
                 <FilterDrawer
+                  onRequestClose={() => {
+                    if (isFilterDrawerSecondaryViewActive) {
+                      setIsFilterDrawerSecondaryViewActive(false);
+                    } else {
+                      setIsFilterDrawerOpen(false);
+                    }
+                  }}
```
Comment (ajaybushel) [inline App.tsx:509]: I think this logic should be internal to the FilterDrawerModal.

Comment (krishnakant-technossus) [reply App.tsx:509]: Actually the states for showing drawer and secondary content are managed outside of this component. That's why I put this logic here.

**Review by raoufr — APPROVED** [review]
Other than @ajaybushel 's comment, looking good to me. Just please add the change log entry for this update.

File: App.tsx (hunk 1 above)
Comment (ajaybushel) [reply App.tsx:509]: got it

**Review by raoufr — APPROVED** [review]

**Pull request merged**